Source/Microsoft.Teams.Apps.CompanyCommunicator.Prep.Func/PreparingToSend/CreateSendingNotificationActivity.cs

```diff
@@ -0,0 +1,72 @@
+﻿// <copyright file="CreateSendingNotificationActivity.cs" company="Microsoft">
+// Copyright (c) Microsoft. All rights reserved.
+// </copyright>
+
+namespace Microsoft.Teams.Apps.CompanyCommunicator.Prep.Func.PreparingToSend
+{
+    using System;
+    using System.Threading.Tasks;
+    using Microsoft.Azure.WebJobs;
+    using Microsoft.Teams.Apps.CompanyCommunicator.Common.Repositories.NotificationData;
+    using Microsoft.Teams.Apps.CompanyCommunicator.Common.Services.AdaptiveCard;
+
+    /// <summary>
+    /// This class contains the "create sending notification data entity" durable activity.
+    /// </summary>
+    public class CreateSendingNotificationActivity
+    {
+        private readonly SendingNotificationDataRepository sendingNotificationDataRepository;
+        private readonly AdaptiveCardCreator adaptiveCardCreator;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="CreateSendingNotificationActivity"/> class.
+        /// </summary>
+        /// <param name="sendingNotificationDataRepository">Sending notification data repository.</param>
+        /// <param name="adaptiveCardCreator">The adaptive card creator.</param>
+        public CreateSendingNotificationActivity(
+            SendingNotificationDataRepository sendingNotificationDataRepository,
+            AdaptiveCardCreator adaptiveCardCreator)
+        {
+            this.sendingNotificationDataRepository = sendingNotificationDataRepository;
+            this.adaptiveCardCreator = adaptiveCardCreator;
+        }
+
+        /// <summary>
+        /// Run the activity.
+        /// </summary>
+        /// <param name="context">Durable orchestration context.</param>
+        /// <param name="notificationDataEntity">Notification data entity.</param>
+        /// <returns>A <see cref="Task"/> representing the asynchronous operation.</returns>
+        public async Task RunAsync(
+            DurableOrchestrationContext context,
+            NotificationDataEntity notificationDataEntity)
+        {
+            await context.CallActivityWithRetryAsync(
+                nameof(CreateSendingNotificationActivity.CreateSendingNotificationAsync),
+                new RetryOptions(TimeSpan.FromSeconds(5), 3),
```

Comment:
This repeats often, define as a constant and describe what the effective retry strategy is, so we don't have to keep looking up what the parameters mean. (5s=?, 3=?)